### ISSUE TITLE
Output formatting and truncation

### ISSUE THREAD

**Issue by SwatiPareek** [neoJINXD/Eternity#34]
Currently we get 16 decimal places output, we must consider if we want this many sigfigs in the output as despite our precision being accurate to 12 decimal places, it may not return the exact expected value for common operations because of all of those sigfigs. Design decision involves determining if we want to give user the ability to adjust how many sigfigs are displayed. 

 For large numbers we automatically show in scientific notation form although we intend on having a special output mode with this form, if giving users this control of determining number of sigfigs displayed and output type, we would need to be consistent with this use case. 

Related issues: Fix precision issues #32
https://github.com/neoJINXD/Eternity/issues/32

**Comment by EmanuelUnity3D** [comment]
I agree with the truncation. We should just use scientific notation whenever the sigfigs are insufficient. I'd avoid having scientific notation as it's own output mode since there will be some situations where it is not necessary and others where it is so we should just force it whenever it is. We could have the number of sigfigs be an option though.

**Comment by EmanuelUnity3D** [comment]
Also, I know that we are doing a lot of stuff on the back-end but this should really be on the front-end of things for flexibility.
